python-checks/src/main/java/org/sonar/python/checks/django/DjangoModelStringFieldCheck.java

```diff
@@ -67,6 +77,36 @@ public void initialize(Context context) {
     });
   }
 
+  private static boolean isNotManaged(List<Statement> modelClassBodyStatements) {
+    return modelClassBodyStatements.stream()
+      // lookup for nested Meta class
+      .filter(ClassDef.class::isInstance)
+      .map(ClassDef.class::cast)
+      .filter(cd -> "Meta".equals(cd.name().name()))
+      .map(ClassDef::body)
+      // lookup for Meta class field assignment
+      .map(StatementList::statements)
+      .flatMap(Collection::stream)
+      .filter(AssignmentStatement.class::isInstance)
+      .map(AssignmentStatement.class::cast)
+      // lookup for managed field assignment
+      .filter(fieldAssignment -> fieldAssignment.lhsExpressions()
+        .stream()
+        .map(lhs -> TreeUtils.firstChild(lhs, Name.class::isInstance)
+          .map(Name.class::cast)
+          .orElse(null))
+        .filter(Objects::nonNull)
+        .map(Name::name)
+        .anyMatch("managed"::equals)
+      )
```

Comment:
This statement is quite long. Maybe we could at least refactor this filter into another function.